### ISSUE TITLE
Contacts in GAL are not Sync

### ISSUE THREAD

**Issue by Floyz** [grahamr975/EWS-Office365-Contact-Sync#10]
Hello !

I'm using the script with O365 and it's working well for me and create a contacts from all users that have mailbox .

But in my GAL I also have contacts (they doesn't require a licence), this is just a contact with a mail and a number, but no mailbox attached, so, in outlook I can browse them in the GAL subsection named "All Contacts" . 

![image](https://user-images.githubusercontent.com/40758026/73543339-079f6c80-4437-11ea-8fbd-3dceb1fdc35c.png)

Theses contacts seems to be avoided / not synced in the folder where the script is creating local contacts.

Is this a known issue ?  did I miss a configuration ? or any plans to implement that also ? 

Regards !

**Comment by grahamr975** [comment]
I've added a 'IncludeNonMailboxContacts' switch to the script. Can you try re-downloading and running with this switch? Let me know how this works.

**Comment by Floyz** [comment]
Hello, 

I have tested, it seems it added contacts for mailboxes present in sharedmailbox with no licences.
But It still seems to not sync contact from the "contact" section of exchange. 

On the Exchange perspective, it does sync entries in the underlined blue on this screen capture, but none on the red, event they got mail, phone number etc ... : 

![image](https://user-images.githubusercontent.com/40758026/73738787-615cab00-4745-11ea-98f5-34efecd83e6b.png)

**Comment by Floyz** [comment]
Hello again, i changed a bit the script to include thoses contacts.

Because the contact list is fetched from Get-User, this list doesn't include contacts, so I needed to append to the contact list variable the return of Get-Contact . 

I added a line at 51 in the Get-GALContacts.ps1 script in EWSContacts\Module\functions\contacts

![image](https://user-images.githubusercontent.com/40758026/73755209-fb345000-4765-11ea-8409-2e68189abb2b.png)

This is working now with this.
Thanks !

**Comment by grahamr975** [comment]
I'll incorporate this as an optional switch. Thanks for finding this.

**Comment by grahamr975** [comment]
I found my previous configuration to lack clarity, so I changed to default behavior to now automatically include shared mailboxes as contacts. Now users will need to use the ExcludeSharedMailboxContacts switch if they need to exclude shared mailboxes.
Now, to include non-user contacts you'll only need to use the IncludeNonUserContacts switch. All of the other changes in this latest update are documented in the change-log file.
![image](https://user-images.githubusercontent.com/56604984/73859580-430bb380-4808-11ea-8f7d-78d379c677ec.png)
